Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+        class_type: Callable,
+        element: Optional[Element],
+):
+    """
+    Turns an XML `Element` Object into an instance of the provided dataclass. Dataclass paramaters must match
+    child XML tags exactly.
+    """
+    d = {}
+    if element is None:
+        return
+
+    if device.hostname:
+        d["hostid"] = device.hostname
+    if device.serial:
+        d["hostid"] = device.serial
+
+    child_element: Element
+    # Handle the XML attributes, if any and if they match dataclass field
+    for attr_name, attr_value in element.attrib.items():
+        dataclass_field = next((x for x in fields(class_type) if x.name == attr_name), None)
+        if dataclass_field:
+            d[attr_name] = attr_value
+
+    d = flatten_xml_to_dict(element, d, class_type)
+
+    return class_type(**d)
+
+
+def resolve_host_id(device: PanDevice):
+    """
+    Gets the ID of the host from a PanDevice object. This may be an IP address or serial number.
+    """
+    host_id: str = ""
+    if device.hostname:
+        host_id = device.hostname
+    if device.serial:
+        host_id = device.serial
+
+    return host_id
+
+
+def resolve_container_name(container: Union[Panorama, Firewall, DeviceGroup, Template, Vsys]):
+    """
+    Gets the name of a given PanDevice container or if it's not a container, returns shared.
+    """
+    if type(container) in [Panorama, Firewall]:
+        return "shared"
+
+    return container.name
+
+
+class PanoramaCommand:
+    """Commands that can only be run, or are relevant only on Panorama."""
+    GET_DEVICEGROUPS_COMMAND = "show devicegroups"
+    GET_TEMPLATE_STACK_COMMAND = "show template-stack"
+
+    @staticmethod
+    def get_device_groups(topology: Topology, device_filter_str: str = None) -> List[DeviceGroupInformation]:
+        """Get all the device groups from Panorama and their associated devices."""
+        device: Panorama
+        result = []
+        for device in topology.active_top_level_devices(device_filter_str):
+            if type(device) is Panorama:
+                response = run_op_command(device, PanoramaCommand.GET_DEVICEGROUPS_COMMAND)
+                for device_group_xml in response.findall("./result/devicegroups/entry"):
+                    dg_name = get_element_attribute(device_group_xml, "name")
+                    for device_xml in device_group_xml.findall("./devices/entry"):
+                        r: DeviceGroupInformation = dataclass_from_element(device, DeviceGroupInformation,
+                                                                           device_xml)
+                        r.name = dg_name
+                        result.append(r)
+
+        return result
+
+    @staticmethod
+    def get_template_stacks(topology: Topology, device_filter_str: str = None) -> List[
+        TemplateStackInformation]:
+        """Get all the template-stacks from Panorama and their associated devices."""
+
+        device: Panorama
+        result = []
```

Comment:
why not just call this variable template_stacks?

Reply:
Easier to name this result, as it's the result of the command. Makes it easier to copy and paste for new functions as well :)